core/src/lib/tls_openssl_private.cc

```diff
@@ -95,6 +125,29 @@ bool TlsOpenSslPrivate::init()
     return false;
   }
 
+  if (!protocol_.empty()) {
+    SSL_CONF_CTX_set_flags(openssl_conf_ctx_,
+                           SSL_CONF_FLAG_FILE | SSL_CONF_FLAG_SHOW_ERRORS |
+                               SSL_CONF_FLAG_CLIENT | SSL_CONF_FLAG_SERVER);
+
+    bool err =
+        SSL_CONF_cmd(openssl_conf_ctx_, "Protocol", protocol_.c_str()) != 2;
```

Comment:
I still believe we're missing `SSL_CONF_finish()`. I'll try to find out why it didn't work for you and keep you updated.